dvc/config.py

```diff
@@ -24,13 +24,29 @@ def __init__(self, msg):
 
 
 class NoRemoteError(ConfigError):
-    def __init__(self, command):
-        msg = (
-            "no remote specified. Setup default remote with\n"
-            "    dvc remote default <name>\n"
-            "or use:\n"
-            "    dvc {} -r <name>\n".format(command)
-        )
+    def __init__(self, command, *, config=None):
+        has_any_remote = False
+        if config:
+            has_any_remote = bool(
+                config.list_options(
+                    Config.SECTION_REMOTE_REGEX, Config.SECTION_REMOTE_URL
+                )
+            )
+
+        if has_any_remote:
+            msg = (
+                "no remote specified. Setup default remote with\n"
+                "    dvc remote default <remote name>\n"
+                "or use:\n"
+                "    dvc {} -r <remote name>".format(command)
+            )
+        else:
+            msg = (
+                "no remote specified. Create a remote with\n"
+                "    dvc remote add <remote name> <remote url>\n"
+                "To create a default remote use the -d flag:\n"
+                "    dvc remote add -d <remote name> <remote url>"
+            )
```

Comment:
Having such a complex logic in the exception is self is rather odd. Why not process config in the code itself and then raise NoRemoteError with the pre-crafted message?

Reply:
Works for me. The message was being generated in `__init__` so I kept it there.